src/client.js

```diff
@@ -136,14 +136,29 @@ export default class StitchClient {
    */
   authenticate(providerType, options = {}) {
     // reuse existing auth if present
-    return this.auth.getAccessToken().then(accessToken => {
+    return Promise.all(
+      [
+        this.auth.getAccessToken(),
+        this.auth.getLoggedInProviderType()
+      ]
+    ).then(([accessToken, loggedInProviderType]) => {
       if (accessToken) {
-        return this.auth.authedId();
+        // TODO: `anon` should be enumerated once providers are maintain
```

Comment:
[nit] Instead of enumeration, we could just have a constant for now

Reply:
We could, but they'd exist in an exported global state. Otherwise we'd have to heavily refactor `providers`

Reply:
Why would it be global if you're just importing the definitions?

Reply:
Right, I guess just because it isn't bound to any class or function doesn't mean it's global if it's exported properly and is stateless. I'm fine with making these constants, but it will be much nicer when each provider is implementing an interface that owns this const.